contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -4418,3 +4418,45 @@ BEGIN
 END;
 $body$
 LANGUAGE plpgsql IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION SYS.TYPE_NAME(IN type_id INT)
+RETURNS SYSNAME AS $$
+    BEGIN
+        RETURN (SELECT ty.name FROM sys.types ty WHERE ty.user_type_id = type_id);
+    END;
+$$
+LANGUAGE plpgsql IMMUTABLE
+STRICT;
+
+CREATE OR REPLACE FUNCTION SYS.TYPE_ID(IN typename SYS.NVARCHAR)
+RETURNS INT AS $$
+    DECLARE
+        schemaid int;
+        schema_name VARCHAR;
+        type_name VARCHAR;
```

Comment:
Rather than using ```VARCHAR```, we should use ```NVARCHAR(128)``` as there is a possibility that user may have a schema or type name with a different collation.